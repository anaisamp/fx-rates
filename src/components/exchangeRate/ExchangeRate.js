import currencies from '../../constants/currencies';

const ExchangeRate = ({ to, from, rates }) => {
  const fromCurrency = currencies.find(c => c.symbol === from).currency;
  const toCurrency = currencies.find(c => c.symbol === to).currency;
  const exchangeRate = (rates[from] && rates[from][to].toFixed(4)) || '...';

  return `${fromCurrency} 1 = ${toCurrency} ${exchangeRate}`;
};

export default ExchangeRate;
