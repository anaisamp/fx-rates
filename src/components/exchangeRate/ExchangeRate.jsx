import React, { Fragment } from 'react';
import currencies from '../../constants';

const ExchangeRate = ({to, from, exchangeRates}) => (
  <Fragment>
    {currencies.find(c => c.symbol === from).currency} 
    1 
    = {currencies.find(c => c.symbol === to).currency} 
    {exchangeRates[from] && exchangeRates[from][to].toFixed(4)}
  </Fragment>
);

export default ExchangeRate;