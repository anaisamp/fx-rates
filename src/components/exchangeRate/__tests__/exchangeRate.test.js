import React from 'react';
import { shallow } from 'enzyme';
import '../../../config/enzyme-setup';
import ExchangeRate from '../ExchangeRate';

it('renders ExchangeRate', () => {
  const props = {
    to: 'GBP',
    from: 'USD',
    rates: { USD: { GBP: 1.46754, EUR: 1.20976 } },
  };
  const wrapper = shallow(<ExchangeRate {...props} />);
  expect(wrapper).toMatchSnapshot();
});
