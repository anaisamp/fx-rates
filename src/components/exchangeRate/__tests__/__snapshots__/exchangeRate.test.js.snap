// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ExchangeRate 1`] = `"$ 1 = Â£ 1.4675"`;
